samples/infix/examples/multi_eq_bad.infix

```diff
@@ -0,0 +1,3 @@
+x = 5 = 6;
```

Comment:
This is not allowed - you're looking at a failure test. The `.expect` file lists the intended syntax errors.

The one remaining case is `x = 5 =;`, which is technically accepted due to the `.seq` implementation.
I can rewrite the rules for that, so even that one is forbidden (and the expected error sequence for this file grows by 1).

Reply:
I tried it, and I ran into something troublesome: the obvious implementation (to me) makes the `expressions` pass not-once.
Worth noting that it originally was, I think, though it wasn't flagged (I know this code was written pre-once originally). It would have to have been for the error rules to not break things.

My pattern looks like this:
```cpp
In(Calculation) *
    (T(Group) << (T(Ident)[Id] * T(Equals) * Any++[Rhs])) >>
  [](Match& _) { return Assign << _(Id) << (Expression << *_[Rhs]); },
```

It reads `<id> = ...` and makes it look like the parse tree we want. The problem is that the nested expression is now part of the rule output, and it must be processed in pass 2. Before that patterns, we match some of unprocessed `Rhs` to error rules and fail out.

It seems that the best way to fix this (well, easiest to understand for me) is to make an `expression_errs` pass that cleans up error cases after `expressions` is done. The WF if I do that, though, will be worse than the `maths` one I had before. I really want to say "just let me mix the two up randomly for a second while I scan for errors", but actually writing that is tricky.

Reply:
TODO: upon discussion, the rule can be move _up_ as opposed to down, which is much easier.

Reply:
Done as of latest commit.